test/implicit/changed-node.py

```diff
@@ -46,13 +46,12 @@
 
 def lister(target, source, env):
     import os
-    with open(str(target[0]), 'w') as ofp:
-        s = str(source[0])
-        if os.path.isdir(s):
-            for l in os.listdir(str(source[0])):
+    with open(target[0], 'w') as ofp:
+        if os.path.isdir(source[0]):
+            for l in os.listdir(source[0]):
                 ofp.write(l + '\\n')
         else:
-            ofp.write(s + '\\n')
+            ofp.write(str(source[0]) + '\\n')
```

Comment:
Why str() these?

Reply:
It's not a file operation... functions like `open` recognize the object has an `__fspath__` and call it, but ordinary string operations like this one can't do that.

Reply:
... an earlier assignment (`s = str(source[0])`) got dropped so the str conversion needed to be put back here